### PR TITLE
NIAD-223: scaffold translator

### PR DESCRIPTION
**Pull request by bartek-sarul**

### CODE REVIEW

**Review by mattd-kainos — DISMISSED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/model/fhir/ParameterNames.java [hunk 1]
```diff
@@ -3,11 +3,10 @@
 import lombok.Getter;
 import lombok.RequiredArgsConstructor;
```
Comment (jaroslaw-wojciechowski) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/model/fhir/ParameterNames.java]: unused imports

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/model/fhir/ParametersExtension.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+@RequiredArgsConstructor
+public class ParametersExtension {
+
+    private final Parameters parameters;
+
+    public static Patient extractPatient(Parameters parameters) {
+        return new ParametersExtension(parameters).extractResource(ParameterNames.PATIENT, Patient.class);
+    }
+
+    public Patient extractPatient() {
```
Comment (jaroslaw-wojciechowski) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/model/fhir/ParametersExtension.java]: for future use or not needed anymore?

Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/model/fhir/ParametersExtension.java]: this one is not static - the static one is actually calling it

Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/model/fhir/ParametersExtension.java]: or at least it should :)

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/FhirToEdifactManager.java [hunk 3]
```diff
@@ -31,15 +36,15 @@
     }
 
     private List<Segment> delegateAcceptance(Parameters parameters) throws FhirValidationException {
-        String acceptanceType = determineAcceptanceType(parameters);
-        if("birth".equals(acceptanceType)) {
-            return acceptanceBirthTranslator.translate(parameters);
+        var acceptanceType = ParametersExtension.extractAcceptanceType(parameters);
```
Comment (jaroslaw-wojciechowski) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/FhirToEdifactManager.java:39]: TODO from 'createMessageSegments' might be removed now?

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/model/fhir/ParametersExtension.java (hunk 2 above)
Comment (bartek-sarul) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/model/fhir/ParametersExtension.java]: will be used in @grzegorz-robak  PR

Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/model/fhir/ParametersExtension.java]: Yeah I just fixed that :D

**Review by jaroslaw-wojciechowski — DISMISSED** [review]

**Review by grzegorz-robak — DISMISSED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/FhirToEdifactManager.java (hunk 3 above)
Comment (bartek-sarul) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/FhirToEdifactManager.java:39]: no, it's needed

**Review by mattd-kainos — APPROVED** [review]

**Pull request merged**